date_range/wizard/date_range_generator.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2016 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from dateutil.relativedelta import relativedelta
+from dateutil.rrule import DAILY, MONTHLY, WEEKLY, YEARLY, rrule
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class DateRangeGenerator(models.TransientModel):
+    _name = 'date.range.generator'
+    _description = 'Date Range Generator'
+
+    @api.model
+    def _default_company(self):
+        return self.env['res.company']._company_default_get('date.range')
+
+    name_prefix = fields.Char('Range name prefix', required=True)
+    date_start = fields.Date(strint='Start date', required=True)
+    type_id = fields.Many2one(
+        comodel_name='date.range.type', string='Type', required=True,
+        domain="['|', ('company_id', '=', company_id), "
+               "('company_id', '=', False)]", ondelete='cascade')
+    company_id = fields.Many2one(
+        comodel_name='res.company', string='Company',
+        default=_default_company)
+    unit_of_time = fields.Selection([
+        (str(YEARLY), 'years'),
+        (str(MONTHLY), 'months'),
+        (str(WEEKLY), 'weeks'),
+        (str(DAILY), 'days')], required=True)
+    duration_count = fields.Integer('Duration', required=True)
+    count = fields.Integer(
+        string="Number of ranges to generate", required=True)
+
+    def _compute_date_ranges(self):
+        self.ensure_one()
+        vals = rrule(freq=self.unit_of_time, interval=self.duration_count,
+                     dtstart=self.date_start,
+                     count=self.count+1)
+        vals = list(vals)
+        date_ranges = []
+        count_digits = len(str(self.count))
+        for idx, dt_start in enumerate(vals[:-1]):
+            date_start = dt_start.date()
+            # always remove 1 day for the date_end since range limits are
+            # inclusive
+            date_end = vals[idx+1].date() - relativedelta(days=1)
+            date_ranges.append({
+                'name': '%s%0*d' % (
+                    self.name_prefix, count_digits, idx + 1),
+                'date_start': date_start,
+                'date_end': date_end,
+                'type_id': self.type_id.id,
+                'company_id': self.company_id.id})
+        return date_ranges
+
+    @api.onchange('company_id')
+    def _onchange_company_id(self):
+        if self.company_id and self.type_id.company_id and \
+                self.type_id.company_id != self.company_id:
+            self._cache.update(
+                self._convert_to_cache({'type_id': False}, update=True))
```

Comment:
`_convert_to_cache()` is renamed to `_update_cache()` in 13.0.
Please check the entire code, `_convert_to_cache()` is used multiple times.